app/jobs/scrape_job.rb

```diff
@@ -0,0 +1,66 @@
+class PostScrapeWorker < ActiveJob::Base
+  queue_as :default
+
+  require 'capybara'
+  require 'capybara/poltergeist'
+
+  include Sidekiq::Worker
+  include Capybara::DSL
+  Capybara.register_driver :poltergeist do |app|
+    Capybara::Poltergeist::Driver.new(app, timeout: 60, js_errors: false)
+  end
+  Capybara.default_driver = :poltergeist
+
+  def perform(new_catalogue)
+    new_catalogue.each do |catalogue_num|
+      begin
+
+        #visit page
+        visit "http://salefinder.com.au/woolworths-catalogue/weekly-specials-catalogue/#{catalogue_num}/catalogue"
+        
+        #creating new catalogue
+        @catalogue = Catalogue.new
+        @catalogue.title = page.all('div#breadcrumb li a')[2].text
+        @catalogue.date = page.find('span.sale-dates').text
+        @catalogue.catalogue_num = catalogue_num
+        @catalogue.shop = page.all('div#breadcrumb li a')[1].text
+        @catalogue.shop.slice!(" catalogues")
+        if @catalogue.shop == "Woolworths"
+          @catalogue.shop_logo = "http://salefinder.com.au/images/retailerlogos/126.jpg"
+        else
+          @catalogue.shop_logo = "http://salefinder.com.au/images/retailerlogos/148.jpg"
+        end
+        @catalogue.save
+
+        #start scraping from page 1 and loop to last page
+        page_num = 1
+        visit "http://salefinder.com.au/woolworths-catalogue/weekly-specials-catalogue/#{catalogue_num}/list?qs=#{page_num}"

```

Comment:
Extract the shared part of the URL into a variable to be used on line 19 and 37.